### PR TITLE
US12001 - Multiple Opportunity End Date

### PR DESCRIPTION
**Pull request by jpc11623**
Commit of smoke tested code

### CODE REVIEW

File: Crossroads.Service.Serve/Controllers/EventController.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        readonly IEventService _eventService;
+        private readonly ILog _logger = LogManager.GetLogger(MethodBase.GetCurrentMethod().DeclaringType);
+
+        public EventController(IEventService eventService)
+        {
+            _eventService = eventService;
+        }
+
+        [HttpGet]
+        [Route("last/{eventid}")]
```
Comment (timmyg) [inline Crossroads.Service.Serve/Controllers/EventController.cs]: should be `{eventid}/last`

File: Crossroads.Service.Serve/Controllers/ResponseController.cs [hunk 2]
```diff
@@ -60,12 +60,12 @@ public IActionResult GetResponsesAndOpportunities([FromQuery] int groupId, [From
         }
 
         [HttpPost]
-        [Route("opportunity/{opportunityId}/event/{eventId}/participant/{participantId}/frequency/{frequencyType}")]
-        public IActionResult CreateMultipleResponses(int opportunityId, int eventId, int participantId, string frequencyType)
+        [Route("opportunity/{opportunityId}/event/{eventId}/participant/{participantId}/frequency/{frequencyType}/end/{endDate}")]
+        public IActionResult CreateMultipleResponses(int opportunityId, int eventId, int participantId, string frequencyType, DateTime endDate)
```
Comment (timmyg) [inline Crossroads.Service.Serve/Controllers/ResponseController.cs]: end date is optional right? wont this break the existing end point if you dont select an end date?
end date could be made a query parameter

File: Crossroads.Service.Serve/Services/Events/EventService.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        {
+            _eventRepository = eventRepository;
+            _mapper = mapper;
+        }
+
+        public EventDto GetLastEventInSeries(int eventId)
+        {
+            var selectedEvent = _eventRepository.GetEvent(eventId);
+            var mpEvents =
+                _eventRepository.GetFutureEventsByType(selectedEvent.EventTypeId, selectedEvent.EventStartDate, null);
```
Comment (timmyg) [inline Crossroads.Service.Serve/Services/Events/EventService.cs]: null param shouldnt be there if optional

File: Crossroads.Service.Serve/Services/Responses/ResponseService.cs [hunk 4]
```diff
@@ -67,11 +67,11 @@ public IList<ResponseDto> CreateSingle(int opportunityId, int eventId, int group
             return null;
         }
 
-        public IList<ResponseDto> CreateMultiple(int opportunityId, int eventId, int participantId, string frequency)
+        public IList<ResponseDto> CreateMultiple(int opportunityId, int eventId, int participantId, string frequency, DateTime endDate)
```
Comment (dtkocher) [inline Crossroads.Service.Serve/Services/Responses/ResponseService.cs]: End date should be optional parameter here.

File: Crossroads.Service.Serve/Controllers/ResponseController.cs (hunk 2 above)
Comment (jpc11623) [reply Crossroads.Service.Serve/Controllers/ResponseController.cs]: End date is not optional for multiple responses, from what I can tell. We default to the last date for that event in a series (it's the one selected in the drop down by default). The single signup hits a different route, so we should always be getting an end date passed down.

**Pull request merged**